### ISSUE TITLE
Establish best practice for plugin config tiddlers

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#2824]
**Problem**
Some, but not all, plugins feature *user settings* in "plugin config tiddlers". There is no *best practice* for plugin authors to present such tiddlers but there are currently two *likely* routes that authors take:

* In the plugin library entry for that plugin, make a *tab*, similar to a *readme*, to display the settings. This is not very good because the end user must keep track of which plugins that have such settings - and they should ideally know which specific tidbits in their TW that are configurable. by plugins 
* Create a separate Controlpanel tab especially made for that one plugin and put this special tab at some arbitrary level, ("Oh, surely *my* plugin is important enough to be a top level tab in the Controlpanel!")

Clearly, neither is very good. 

**Proposal**
Establish a "best practice" for where plugins should display their plugin config tiddlers, like so:

1. Introduce a separate tab titled e.g *Plugin settings* in the Controlpanel tab hierarchy that lists all, and only, "plugin setting tiddlers".  For this to work, there should probably be a system tag e.g `$:/tags/PluginSettings`. 
2. Such "plugin setting tiddlers" could be grouped by "plugin name path" (e.g `$:/plugins/TWaddle/...`) or by author name. The point is to have all settings from one author collected. I believe either the ToC macro or the experimental [System tiddler browser](http://tiddlywiki.com/prerelease/#%24%3A%2Fcore%2Fui%2FMoreSideBar%2FExplore) could be used to group and present these setting tiddlers .

**Comment by danielo515** [comment]
I like the idea .

Giving guidelines is a good thing, but do you know what is better ? Give tools to make following such guides easier. 
I'm creating a yeoman generator for tw5 plugin development . Maybe you could help me with it ?

**Comment by Jermolene** [comment]
Hi @twMat 

I agree that this is an area that could do with stronger conventions.

On your specific proposals, the first sounds a lot like what we have today in the "Plugin" tab: a place where plugins are listed, and their  documentation and settings can be accessed.

A standard place for plugin docs and settings makes them easily findable, but prevents settings from being located where it's logical. For example, in 5.1.14, the "Saving" tab of control panel is extensible, and saver plugins can add their own configuration pages.

So, I think there's a basic tension between the desire to group settings together under the plugin/author to which they relate versus grouping settings in logical categories according to function. We need to support both.

Anyhow, I'm interested to see any proposals/prototypes.

**Comment by tobibeer** [comment]
To me it sounds rather simple for plugins to provide a "settings" tiddler just like "readme" and displayed the same... which can equally be aggregated to some "plugin settgins" listing.

**Comment by twMat** [comment]
@danielo515 
>I'm creating a yeoman generator for tw5 plugin development . Maybe you could help me with it ?

Contact me privately. I have a pretty cool idea for this that you might like.

**Comment by twMat** [comment]
@Jermolene 

>Anyhow, I'm interested to see any proposals/prototypes.

[demo](http://pluginsettings.tiddlyspot.com/)

Under Ctrlpanel > Settings, I made the standard settings content into a tab plus added two extra tabs to show two variants for "Plugin settings". It should really only be one *Plugin settings* tab, I'm just illustrating a difference between *author name* and *brand*. 

Note that for the demo I've used ".../readme" as content because there are none with ".../settings".

Layout: I should probably have the uthor/brand as buttons rather than the individual plugins. Or maybe both, not sure.

**Regarding the OP**
In this demo, the called for "best practice" is solved by asking the user to submit a `$:/plugins/..../settings` tiddler (**IF** they want to include plugin settings). This is comparable to ask them to submit a `../readme`. Thus no need for new system tags.

If the setting tiddlers are grouped by **brand**, as extracted from the plugin name, this can be automatic and the plugin-maker deosn't have to provide an author name.

--- 
Side note: I fail with the `<<qualify>>` macro when trying to qualify a variable. Could someone help?
